src/unix/unix_c/unix_tcsetattr_job.c

```diff
@@ -34,8 +34,12 @@ static void worker_tcsetattr(struct job_tcsetattr *job)
         job->result = result;
         job->error_code = errno;
     } else {
-        decode_terminal_status(&termios, &(job->termios[0]));
-        job->result = tcsetattr(job->fd, job->when, &termios);
+        int result_decode = decode_terminal_status(&termios, &(job->termios[0]));
+        if (result_decode < 0) {
```

Comment:
I suggest to change this comparison to `!= 0`, and return `EINVAL` rather than `-EINVAL` from `decode_terminal_status`. Set `job->result` to `-1` in the then-branch, as we don't use that value beyond checking that it's negative. We take the error result from `error_code`. `-EINVAL` appears fragile.

Reply:
Sure, that makes sense. Amended the commit!